odb_extractor_atoms.py

```diff
@@ -59,98 +223,297 @@ def select_obsclass(classes):
+        for log_entry in observation['obsLog']:
+            obslog[log_entry['label']] = {'qaState': log_entry['qaState'], 'filename': log_entry['filename']}
+        datalabels = list(obslog.keys())
 
     # Sequence analysis
     sequence = observation['sequence']
     nsteps = len(sequence)
 
+    # First pass to analyze offsets and exptimes/coadds
+    do_not_split = False
+    exptimes = []
+    coadds = []
+    qoffsets = []
+    poffsets = []
+    qoffsets_sky = []
+    poffsets_sky = []
+    config = {'inst': '', 'fpu': [], 'disperser': [], 'filter': [], 'wavelength': []}
+    for ii, step in enumerate(sequence):
+        step_keys = list(step.keys())
+
+        inst = step['instrument:instrument']
+        #     print(inst, fpuinst[inst])
+        if inst == 'Visitor Instrument':
+            inst = step['instrument:name'].split(' ')[0]
+            if inst in ["'Alopeke", "Zorro"]:
+                fpu = 'None'
+            else:
+                fpu = inst
+        else:
+            fpu = step[fpuinst[inst]]
+        config['inst'] = inst
+        config['fpu'].append(fpu)
+
+        if 'instrument:disperser' in step_keys:
+            disperser = step['instrument:disperser']
+        elif inst in ['IGRINS', 'MAROON-X']:
+            disperser = inst
+        else:
+            disperser = 'None'
+        if inst == 'GNIRS':
+            if step['instrument:acquisitionMirror'] == 'in' and \
+                    step['instrument:decker'] == 'acquisition':
+                disperser = 'mirror'
+            else:
+                disperser = disperser.strip('grating') + step['instrument:crossDispersed']
+        elif inst == 'Flamingos2' and fpu == 'FPU_NONE':
+            if step['instrument:decker'] == 'IMAGING':
+                disperser = step['instrument:decker']
+        config['disperser'].append(disperser)
+
+        if 'instrument:filter' in step_keys:
+            filter = step['instrument:filter']
+        elif inst == 'GPI':
+            filter = find_filter(fpu, gpi_filter_wav)
+        else:
+            if inst == 'GNIRS':
+                filter = 'None'
+            else:
+                filter = 'Unknown'
+        if inst == 'NIFS' and 'Same as Disperser' in filter:
+            for filt in list(nifs_filter_wav.keys()):
+                if disperser[0] in filt:
+                    filter = filt
+                    break
+        config['filter'].append(filter)
+
+        if inst == 'GPI':
+            wavelength = gpi_filter_wav[filter]
+        else:
+            wavelength = float(step['instrument:observingWavelength'])
+        config['wavelength'].append(wavelength)
+
+        p = 0.0
+        q = 0.0
+        # Just want exposures on sky for dither pattern analysis
+        if step['observe:observeType'].upper() not in ['FLAT', 'ARC', 'DARK', 'BIAS']:
+            if 'telescope:p' in step_keys:
+                p = float(step['telescope:p'])
+            else:
+                p = 0.0
```

Comment:
Repeated strings like `telescope:p` should probably be defined somewhere since "magic strings" like this are considered to be a bad idea.

While it isn't strictly necessary, this could all be simplified to:
```
p = float(step['telescope:p']) if 'telescope:p' in step else 0.0
```